addons/binding/org.openhab.binding.phc/src/main/java/org/openhab/binding/phc/handler/PHCBridgeHandler.java

```diff
@@ -0,0 +1,710 @@
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.StopMoveType;
+import org.eclipse.smarthome.core.library.types.UpDownType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.phc.PHCBindingConstants;
+import org.openhab.binding.phc.internal.PHCHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import gnu.io.CommPortIdentifier;
+import gnu.io.PortInUseException;
+import gnu.io.RXTXPort;
+import gnu.io.SerialPort;
+import gnu.io.SerialPortEvent;
+import gnu.io.SerialPortEventListener;
+import gnu.io.UnsupportedCommOperationException;
+
+/**
+ * The {@link PHCBridgeHandler} is responsible for handling the serial Communication to and from the PHC Modules.
+ *
+ * @author Jonas Hohaus - Initial contribution
+ */
+
+public class PHCBridgeHandler extends BaseBridgeHandler implements SerialPortEventListener {
+
+    private Logger logger = LoggerFactory.getLogger(PHCHandler.class);
+
+    private static final int BAUD = 19200;
+    private static final int SEND_RETRY_COUNT = 15; // max count to send the same message
+    private static final int SEND_RETRY_TIME = 80; // time to wait for an acknowledge before send the message again
+
+    private InputStream serialIn = null;
+    private OutputStream serialOut = null;
+    short lastReceivedCrc;
+    byte[] messageFragment = null;
+    RXTXPort serialPort = null;
+
+    private Map<String, Boolean> toggleMap = new HashMap<String, Boolean>();
+    private ConcurrentSkipListMap<Long, QueueObject> queue = new ConcurrentSkipListMap<Long, QueueObject>();
+
+    private byte emLedOutputState[] = new byte[32];
+    private byte amOutputState[] = new byte[32];
+
+    public PHCBridgeHandler(Bridge phcBridge) {
+        super(phcBridge);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing PHC Bridge.");
+
+        String port = ((String) getConfig().get(PHCBindingConstants.PORT));
+        logger.debug(port);
+
+        try {
+            CommPortIdentifier portId = null;
+
+            // parse ports and if the default port is found, initialized the
+            // reader
+            @SuppressWarnings("rawtypes")
+            Enumeration portList = CommPortIdentifier.getPortIdentifiers();
+            while (portList.hasMoreElements()) {
+                CommPortIdentifier id = (CommPortIdentifier) portList.nextElement();
+                if (id.getPortType() == CommPortIdentifier.PORT_SERIAL) {
+                    if (id.getName().equals(port)) {
```

Comment:
I think that example is not the most optimal, our codebase also contains the following which should be prefered (if it works ;-))

```java
CommPortIdentifier.getPortIdentifier(portName);
```